host/xtest/regression_1000.c

```diff
@@ -3272,3 +3272,37 @@ static void xtest_tee_test_1039(ADBG_Case_t *c)
 
 ADBG_CASE_DEFINE(regression, 1039, xtest_tee_test_1039,
 		 "Test subkey verification");
+
+static void xtest_tee_test_1040(ADBG_Case_t *c)
+{
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	TEEC_Session session = { };
+	uint32_t ret_orig = 0;
+
+	res = xtest_teec_open_session(&session, &pta_invoke_tests_ta_uuid, NULL,
+				      &ret_orig);
+	if (res == TEEC_ERROR_ITEM_NOT_FOUND) {
+		/* Pseudo TA is optional: warn and nicely exit */
+		Do_ADBG_Log(" - 1040 -   skip test, pseudo TA not found");
+		return;
+	}
+	ADBG_EXPECT_TEEC_SUCCESS(c, res);
+
+	res = TEEC_InvokeCommand(&session, PTA_INVOKE_TESTS_CMD_ITR_NOTIF_TESTS,
+				 NULL, &ret_orig);
+	if (res != TEEC_SUCCESS) {
+		ADBG_EXPECT_TEEC_ERROR_ORIGIN(c, TEEC_ORIGIN_TRUSTED_APP,
+					      ret_orig);
+		if (res == TEEC_ERROR_NOT_SUPPORTED) {
+			/* Embedding tests is optional: warn and nicely exit */
+			Do_ADBG_Log(" - 1040 -   skip test, not implemented");
```

Comment:
If the `ADBG_EXPECT_TEEC_ERROR_ORIGIN` call fails, then this doesn't help, or? The test will still end with an error?

Reply:
This test calls the Invoke test PTA. Line 3284 already address the case where the PTA is not embedded. Once we know the PTA is embedded, we require the PTA to answer our request, hence the test at line 3294.
Di I miss something?

Reply:
After re-reading the code, I think you're right. My first impression was that this eventually could end up logging an adbg error, while still printing the "skip test". 
`Acked-by: Joakim Bech <joakim.bech@linaro.org>`